tests/python/relax/test_codegen_cudnn.py

```diff
@@ -36,12 +36,8 @@ def reset_seed():
 
 has_cudnn = tvm.get_global_func("relax.ext.cudnn", True)
 
-cudnn_enabled = pytest.mark.skipif(
-    not has_cudnn,
-    reason="cuDNN not enabled.",
-)
 
-pytestmark = [cudnn_enabled]
+pytestmark = [*tvm.testing.requires_cudnn.marks()]
```

Comment:
Nitpick: The `tvm.testing.requires_*` marks can be used as marks directly, rather than needing to be explicitly expanded with `foo.marks()`.  In addition, the `pytestmark` can be a single marker, rather than a list.  Between these two, this can cleaned up as:

```python
pytestmark = tvm.testing.requires_cudnn
```

Reply:
I'm not sure that it's possible. Global object `pytestmark` should has type `Mark` or `List[Mark]` but our TVM markers has type `tvm.testing.utils.Feature`. So direct assignment, like you suggest above, will lead to type check error:
> TypeError: got <tvm.testing.utils.Feature object at 0x7ff96f507340> instead of Mark

Moreover, the idea of this change was to utilise hierarchical structure of tvm testing features. `pytest.mark.gpu` mark is a root of this hierarchy. So I have to extract all markers from current feature and all parents. That is exactly method `marks()` do.

But you are right this line can be slightly simplified by removing explicit list generator, like:
> pytestmark = tvm.testing.requires_cudnn.marks()

Reply:
Good point.  I thought I had made `Feature` be a subclass of `Mark`, but I guess I never did.  In that case, I like your simplification.